onmt/dynamic/shard.py

```diff
@@ -0,0 +1,193 @@
+"""Module that contain shard utils for dynamic data."""
```

Comment:
Is this whole sharding thing really necessary?
Can't we just directly iterate on chunks of each file? Would save some time and disk space.

Reply:
It does look a bit redundant and didn't do much work. I'll try to remove this part latter to see if this causes any issue.